pkg/controllers/automigration/controller.go

```diff
@@ -301,24 +301,35 @@ func (c *Controller) estimateCapacity(
 	estimatedCapacity := make(map[string]int64, len(clusterObjs))
 
 	for _, clusterObj := range clusterObjs {
-		unsObj := clusterObj.Object.(*unstructured.Unstructured)
+		keyedLogger := keyedLogger.WithValues("cluster", clusterObj.ClusterName)
+
+		unsClusterObj := clusterObj.Object.(*unstructured.Unstructured)
 
 		// NOTE: these should follow Deployment's status semantics and should not include terminating pods
-		totalReplicas, readyReplicas, err := c.getTotalAndReadyReplicas(unsObj)
+		//
+		// TODO: what if some workloads do not have a field that represents the totalReplicas or readyReplicas?
+		totalReplicas, readyReplicas, err := c.getTotalAndReadyReplicas(unsClusterObj)
 		if err != nil {
-			keyedLogger.Error(err, "Failed to get total and ready replicas from object", "cluster", clusterObj.ClusterName)
+			// TODO: since we are now using desiredReplicas - unschedulablePods to calculate estimatedCapacity,
```

Comment:
Why is this a TODO?

Reply:
It's related to the TODO above. Will merge them together.